### PR TITLE
Add an ability to read noncontinuous RecordIO and TFRecord files

### PR DESCRIPTION
**Pull request by JanuszL**
- adds an ability to seek to index file loader if the next entry is not next to the previous one
- adds a test

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It adds an ability to seek to index file loader if the next entry is not next to the previous one

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     adds an ability to seek to index file loader if the next entry is not next to the previous one
 - Affected modules and functionalities:
     index_file loader
 - Key points relevant for the review:
     NA
 - Validation and testing:
     new test added
 - Documentation (including examples):
     No

In response to https://github.com/NVIDIA/DALI/issues/1744

**JIRA TASK**: *[NA]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

File: dali/operators/reader/loader/recordio_loader.h [hunk 1]
```diff
@@ -100,10 +100,11 @@ class RecordIOLoader : public IndexedFileLoader {
       return;
     }
 
-    if (should_seek_) {
+    if (should_seek_ || next_seek_pos_ != seek_pos) {
       current_file_->Seek(seek_pos);
       should_seek_ = false;
     }
+    next_seek_pos_ = seek_pos + size;
```
Comment (mzient) [inline dali/operators/reader/loader/recordio_loader.h]: This is very brittle and may lead to very subtle and hard-to-detect errors - we have not moved the file pointer yet.

Comment (mzient) [reply dali/operators/reader/loader/recordio_loader.h]: This should be in lines:
120, 135
Also, it looks like line 141 we should reset next_seek_pos_ to 0   ?

Comment (JanuszL) [reply dali/operators/reader/loader/recordio_loader.h]: Done

**Comment by JanuszL** [comment]
!build

**Pull request merged**